sycl/test-e2e/lit.cfg.py

```diff
@@ -805,6 +767,49 @@ def open_check_file(file_name):
     else:
         config.intel_driver_ver[sycl_device] = {}
 
+if "hip:gpu" in config.sycl_devices and config.hip_platform == "AMD":
+    if not config.amd_arch:
+        amd_arch_prefix = "arch-amd_gpu_"
+        amd_device_arch = [i for i in config.sycl_dev_features["hip:gpu"] if amd_arch_prefix in i]
+        if len(amd_device_arch) == 0:
+            lit_config.error(
```

Comment:
i had to move this block down so the `sycl-ls` parsing we already do we available, but that meant i had to add a loop at the end to update the features